packages/core/helper-plugin/src/content-manager/CMEditViewDataManager.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+
+import { MessageDescriptor } from 'react-intl';
+
+/**
+ * TODO: All these types could be accurately inferred from
+ * their usage in the content-manager when we move it back.
+ */
+interface CMEditViewDataManagerContext {
+  addComponentToDynamicZone?: (
+    keys: string[],
+    componentLayoutData: Record<string, unknown>,
+    allComponents: Record<string, unknown>,
+    shouldCheckErrors?: boolean,
+    position?: number
+  ) => void;
+  addNonRepeatableComponentToField?: (
+    keys: string[],
+    componentLayoutData: Record<string, unknown>,
+    allComponents: Record<string, unknown>
+  ) => void;
+  addRepeatableComponentToField?: (
+    keys: string[],
+    componentLayoutData: Record<string, unknown>,
+    allComponents: Record<string, unknown>,
+    shouldCheckErrors?: boolean,
+    position?: number
+  ) => void;
+  allLayoutData: {
+    // TODO: this should be a schema?
+    components: Record<string, unknown>;
+    // TODO: this should be a schema?
+    contentType: Record<string, unknown>;
```

Comment:
I think the types could be improved by extending [schema](https://github.com/strapi/strapi/blob/main/packages/core/strapi/lib/types/core/schemas/index.d.ts), I see you have the TODOs but maybe we could improve it now, what do you think is best?

Reply:
aha, i didn't realise we had them, i'll do some investigation to ensure they're _exactly_ what we want, but nice catch ✨